### ISSUE TITLE
What's the use of your level set method here

### ISSUE THREAD

**Issue by 675492062** [ahatamiz/dals#3]
Hello, Hatamizadeh
What's the use of your level set method here? Is it just post-processing? Seeing your code implementation, the loss during training has nothing to do with level set, and during inference the level set (contour) obtained  is only used to calculate a metric for evaluation. I was so confused for these question.

best wishes

**Comment by ahatamiz** [comment]
In this work we propose an integrated framework (DALS) for combining CNNs with levelsets. The classical active contours usually require some form of interaction with a user for contour and parameter initialization. However, DALS automates the process and directly initializes the levelset from the output of a CNN. Moreover, the DALS is highly optimized and can be scaled for batched inference in Tensorflow. The levelset is not engaged during training. 

One immediate benefit of DALS is its capability to produce more refined boundaries in comparison to CNNs, whose output maybe noisy specially around the edges. In its current format, it is backbone-agnostic ( e.i. can be integrated with any well-trained CNN).

During inference, we report the metric (DICE) on the output of levelset. 

I hope you find DALS useful in your research/project.

**Comment by 675492062** [comment]
Thank you for your reply
     In fact, the output (probs map) of CNNs is to provide the initial Phi for the level set. Lamda 1 and lamda 2 are generally set to 1 in the original level set algorithm. What's the point of constructing two reciprocal numbers to replace them (Lamda 1 and lamda 2 )? puzzled!

best wishes

**Comment by ahatamiz** [comment]
Yes, the CNN provides an initialization. Regarding lambda maps, various papers have shown that simply setting them to 1 is sub-optimal. Traditionally, researchers optimized these values by trial and error. We came up with an analytical formula to determine them in a pixel-wise manner (hence more control over the contour evolution). We have provided more details in the manuscript of our work.

**Comment by 675492062** [comment]
Thanks a lot, Thank you for answering my questions. I see

**Comment by zhaozhiyi11** [comment]
> 非常感谢， 谢谢你回答我的问题。明白了

请问怎样测试训练的模型？

**Comment by Rimehdaoudi** [comment]
hello, 
i have a question about the dataset,is the train folder contains many images and labels each one of them in .npy format() or all the images in one npy file and all labels in one npy file.

**Comment by Rimehdaoudi** [comment]
> Hello, Hatamizadeh What's the use of your level set method here? Is it just post-processing? Seeing your code implementation, the loss during training has nothing to do with level set, and during inference the level set (contour) obtained is only used to calculate a metric for evaluation. I was so confused for these question.
> 
> best wishes

hello, 
i have a question about the dataset,is the train folder contains many images and labels each one of them in .npy format() or all the images in one npy file and all labels in one npy file.
